website/project/model.py

```diff
@@ -2991,7 +3027,8 @@ def add_contributor(self, contributor, permissions=None, visible=True,
             if save:
                 self.save()
 
-            project_signals.contributor_added.send(self, contributor=contributor, auth=auth)
+            all_global_subscriptions_none = check_if_all_global_subscriptions_are_none(contributor)
```

Comment:
It would be good if we could move as much notification code out of the project model as possible. This check, for example, can happen in the listener (https://github.com/CenterForOpenScience/osf.io/blob/6e449d62a128163a6e9519a5d51ca2ed28ccf503/website/project/views/contributor.py#L465) rather than passing it to the signal.